sycl/include/CL/sycl/detail/pi.h

```diff
@@ -46,6 +46,7 @@
 // elsewhere, e.g. in the pi_opencl, but constants/enums mapping is now
 // done here, for efficiency and simplicity.
 //
+#include "export.hpp"
```

Comment:
Should be
```suggestion
#include <CL/sycl/detail/export.hpp>
```

Thing that still bothers me is why we need to change this file at all? The plugin exports functions with prefixes, not these functions. Neither users are supposed to link against these symbols. Can you show an example of linker error that forced you to add these exports?

Reply:
On Linux I did not need __SYCL_EXPORT in pi.h file.
Visual Studio complained that the pi* functions declared in pi.h is different than the ones defined in pi_opencl.cpp, where pi.h is #include-d. 
That's why added __SYCL_EXPORT for pi* functions in this file.

Reply:
Ok, I see. So, since you've changed the symbol names, VS now consideres both of symbols in pi.h and pi_opencl.cpp the same. But I'm not sure they really are. I think we should be careful here, as pi.h is used both in libpi_opencl.so and libsycl.so, and we can accidentally mess up something (like, exporting a pi* symbol from libsycl.so).

Reply:
@smaslov-intel ,
Could you answer this Alexander's question?
My understanding is that we should match the func declation signatures in pi.h with functions defined in pi_opencl.cpp.
On Windows, due to these differences, I had to add __SYCL_EXPORT in pi.h.

Reply:
The pi.h keeps forward declaration of PI API. Plugins (libpi_opencl.so) implement the PI API, so yes they are same symbols. They are not defined by lybsycl.so, only by plugins. And declarations should match definitions.